### PR TITLE
 Quick and dirty fix of the unused_braces lint

### PR DESCRIPTION
**Pull request by flip1995**
cc @lcnr 

Adresses #70814 

This at least prevents lint output, if no span is available. Even though this also prevents the `unused_parens` lint from emitting, when the `DUMMY_SP` is used there, but I think that should be ok, since error messages without a span are quite useless anyway.

Clippy CI is currently blocked on this bug. If this quick and dirty fix should be rejected, I could try to work around this in Clippy.

r? @shepmaster

### CODE REVIEW

File: src/librustc_lint/unused.rs [hunk 1]
```diff
@@ -395,6 +395,12 @@ trait UnusedDelimLint {
         left_pos: Option<BytePos>,
         right_pos: Option<BytePos>,
     ) {
+        // FIXME(flip1995): Quick and dirty fix for #70814. This should be fixed in rustdoc
```
Comment (lcnr) [inline src/librustc_lint/unused.rs:398]: is this actually required? I would expect the check in `emit_unused_delims` to be enough

**Comment by lcnr** [comment]
Sounds good! I considered doing the same in the next few days as I was unable to fix the underlying problem.

We may want to backport this to beta IMO.

File: src/librustc_lint/unused.rs (hunk 1 above)
Comment (flip1995) [reply src/librustc_lint/unused.rs:398]: Oh yeah, you're right. I will change it.

**Comment by flip1995** [comment]
@rust-lang/compiler Maybe a candidate for a beta backport. This crashed a few CIs including Clippy's

**Comment by oli-obk** [comment]
@bors r+ p=1

this is blocking tools and very annoying to users. Merging this so we have time to come up with a nice fix.

**Comment by pnkfelix** [comment]
discussed in T-compiler [meeting](https://zulip-archive.rust-lang.org/131828tcompiler/05709weeklymeeting2020043054818.html#195843962). beta backport approved.

**Pull request merged**